src/openlcb/MemoryConfigClient.hxx

```diff
@@ -402,7 +405,8 @@ private:
 
     Action read_response_timeout()
     {
-        if (responseCode_ & DatagramClient::OPERATION_PENDING)
+        if (responseCode_ & DatagramClient::OPERATION_PENDING ||
+            !timer_.is_triggered())
```

Comment:
if the read finishes upon the first attempt, the timer is never started.
If this code is really needed, then we have to call timer_.set_triggered() on that path.